files/scripts/bgp.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+SERVICE="bgp"
+DEBUGLOG="/tmp/bgp_debug.log"
+LOCKFILE="/tmp/bgp-lock"
+
+function debug()
+{
+		/usr/bin/logger $1
+		/bin/echo `date` "- $1" >> ${DEBUGLOG}
+}
+
+function lock_service_state_change()
+{
+    debug "Locking ${LOCKFILE} from ${SERVICE} service"
+
+    exec {LOCKFD}>${LOCKFILE}
+    /usr/bin/flock -x ${LOCKFD}
+    trap "/usr/bin/flock -u ${LOCKFD}" 0 2 3 15
+
+    debug "Locked ${LOCKFILE} (${LOCKFD}) from ${SERVICE} service"
+}
+
+function unlock_service_state_change()
+{
+    debug "Unlocking ${LOCKFILE} (${LOCKFD}) from ${SERVICE} service"
+    /usr/bin/flock -u ${LOCKFD}
+}
+
+function check_warm_boot()
+{
+    STATE_DBID_PORT=`python -c 'from swsssdk import _get_redis_dbid_port; print _get_redis_dbid_port("STATE_DB")'`
+    SYSTEM_WARM_START=`/usr/bin/redis-cli $STATE_DBID_PORT hget "WARM_RESTART_ENABLE_TABLE|system" enable`
+    SERVICE_WARM_START=`/usr/bin/redis-cli $STATE_DBID_PORT hget "WARM_RESTART_ENABLE_TABLE|${SERVICE}" enable`
+    if [[ x"$SYSTEM_WARM_START" == x"true" ]] || [[ x"$SERVICE_WARM_START" == x"true" ]]; then
+        WARM_BOOT="true"
+    else
+        WARM_BOOT="false"
+    fi
+}
+
+function validate_restore_count()
+{
+    if [[ x"$WARM_BOOT" == x"true" ]]; then
+        STATE_DBID_PORT=`python -c 'from swsssdk import _get_redis_dbid_port; print _get_redis_dbid_port("STATE_DB")'`
+        RESTORE_COUNT=`/usr/bin/redis-cli $STATE_DBID_PORT hget "WARM_RESTART_TABLE|orchagent" restore_count`
```

Comment:
here is what we used internally, for community version, we should use sonic-db-cli